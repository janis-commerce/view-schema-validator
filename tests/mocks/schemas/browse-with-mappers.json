{
  "service": "tms",
  "name": "route-browse",
  "root": "Browse",
  "autoRefresh": true,
  "canCreate": true,
  "canPreview": false,
  "canRefresh": true,
  "canView": false,
  "pageSize": 60,
  "source": {
    "service": "sac",
    "namespace": "claim-type",
    "method": "browse",
    "resolve": false
  },
  "fields": [
    {
      "name": "date",
      "component": "Text",
      "mapper": {
        "name": "date",
        "props": {
          "incomingFormat": "DD/MM/YYYY",
          "format": "DD/MM/YYYY"
        }
      }
    },
    {
      "name": "schedule",
      "component": "Text",
      "mapper": "dateDelivery"
    },
    {
      "name": "incomingDate",
      "component": "TimeChip",
      "mapper": {
        "name": "dateDelivery",
        "props": {
          "incomingFormat": "dd/MM/yyyy HH:mm"
        }
      },

      "componentAttributes": {}
    },
    {
      "name": "nameTest",
      "component": "Text",
      "mapper": {
        "name": "prefix",
        "props": {
          "value": "common.names."
        }
      }
    },
    {
      "name": "currencyTest",
      "component": "Text",
      "mapper": {
        "name": "currency",
        "props": {
          "currencyCode": "USD",
          "currencyField": "someField"
        }
      }
    },

    {
      "name": "name",
      "component": "LightText",
      "filter": {
        "component": "Input"
      },
      "mapper": {
        "name": "suffix",
        "props": {
          "value": "%"
        }
      }
    },
    {
      "name": "appliesToLogistics",
      "component": "Chip",
      "mapper": "booleanToWord"
    }
  ]
}
