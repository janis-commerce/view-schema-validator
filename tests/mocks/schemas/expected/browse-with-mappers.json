{
  "service": "tms",
  "name": "route-browse",
  "root": "Browse",
  "autoRefresh": true,
  "canCreate": true,
  "canPreview": false,
  "canRefresh": true,
  "canView": false,
  "pageSize": 60,
  "source": {
    "service": "sac",
    "namespace": "claim-type",
    "method": "browse",
    "resolve": false
  },
  "filters": [],
  "fields": [
    {
      "name": "date",
      "component": "Text",
      "attributes": {
        "initialSortDirection": "desc",
        "isDefaultSort": false,
        "isStatus": false,
        "sortable": false
      },
      "componentAttributes": {
        "fontWeight": "normal"
      },
      "mapper": {
        "name": "date",
        "props": {
          "incomingFormat": "DD/MM/YYYY",
          "format": "DD/MM/YYYY"
        }
      }
    },
    {
      "name": "schedule",
      "component": "Text",
      "attributes": {
        "initialSortDirection": "desc",
        "isDefaultSort": false,
        "isStatus": false,
        "sortable": false
      },
      "componentAttributes": {
        "fontWeight": "normal"
      },
      "mapper": "dateDelivery"
    },
    {
      "name": "incomingDate",
      "component": "TimeChip",
      "attributes": {
        "initialSortDirection": "desc",
        "isDefaultSort": false,
        "isStatus": false,
        "sortable": false
      },
      "componentAttributes": {},
      "mapper": {
        "name": "dateDelivery",
        "props": {
          "incomingFormat": "dd/MM/yyyy HH:mm"
        }
      }
    },

    {
      "name": "nameTest",
      "component": "Text",
      "attributes": {
        "initialSortDirection": "desc",
        "isDefaultSort": false,
        "isStatus": false,
        "sortable": false
      },
      "componentAttributes": {
        "fontWeight": "normal"
      },
      "mapper": {
        "name": "prefix",
        "props": {
          "value": "common.names."
        }
      }
    },
    {
      "name": "currencyTest",
      "component": "Text",
      "attributes": {
        "initialSortDirection": "desc",
        "isDefaultSort": false,
        "isStatus": false,
        "sortable": false
      },
      "componentAttributes": {
        "fontWeight": "normal"
      },
      "mapper": {
        "name": "currency",
        "props": {
          "currencyCode": "USD",
          "currencyField": "someField"
        }
      }
    },
    {
      "name": "name",
      "component": "LightText",
      "attributes": {
        "initialSortDirection": "desc",
        "isDefaultSort": false,
        "isStatus": false,
        "sortable": false
      },
      "componentAttributes": {
        "fontWeight": "light"
      },
      "filter": {
        "component": "Input",
        "componentAttributes": {},
        "remote": false,
        "type": "equal"
      },
      "mapper": {
        "name": "suffix",
        "props": {
          "value": "%"
        }
      }
    },
    {
      "name": "appliesToLogistics",
      "component": "Chip",
      "attributes": {
        "initialSortDirection": "desc",
        "isDefaultSort": false,
        "isStatus": false,
        "sortable": false
      },
      "componentAttributes": {
        "borderColor": "grey"
      },
      "mapper": "booleanToWord"
    }
  ]
}
